### ISSUE TITLE
Question on DoReFa's fw(.)

### ISSUE THREAD

**Issue by a-maci** [ppwwyyxx/tensorpack#76]
Line 27 and 28 in dorefa.py do:
E = tf.stop_gradient(tf.reduce_mean(tf.abs(x)))
return tf.sign(x / E) * E 

Equation 7 in the DoReFa paper says: 
Forward: ro = sign(ri) × E(|ri|)

Question: Equation in the paper does not normalize the weights before taking the sign but the code does this. Is this OK?

**Comment by ppwwyyxx** [comment]
Normalize by a factor doesn't change the sign at all.

This normalization is only to make auto-differentiation be consistent with the paper.
